common/oatbox/filesystem/Directory.php

```diff
@@ -137,7 +136,13 @@ public function exists()
      */
     public function deleteSelf()
     {
-        return $this->getFileSystem()->deleteDir($this->getPrefix());
+        try {
+            $this->getFileSystem()->deleteDirectory($this->getPrefix());
+            return true;
+        } catch (FilesystemException $e) {
+        }
```

Comment:
Yes, it would be certainly useful when debugging, the response might be just about a missing object, but could also be something related to connection, visibility, etc. Updated here and elsewhere